test/validation/api/ipsec/ipsec.c

```diff
@@ -1119,9 +1222,9 @@ int ipsec_init(odp_instance_t *inst)
 	if (suite_context.inbound_op_mode == ODP_IPSEC_OP_MODE_ASYNC ||
 	    suite_context.outbound_op_mode == ODP_IPSEC_OP_MODE_ASYNC ||
 	    suite_context.outbound_op_mode == ODP_IPSEC_OP_MODE_INLINE) {
-		out_queue = odp_queue_create("ipsec-out", NULL);
-		if (ODP_QUEUE_INVALID == out_queue) {
-			fprintf(stderr, "IPsec outq creation failed.\n");
+		default_queue = odp_queue_create("ipsec-default", NULL);
```

Comment:
If plain queues are not supported as destination queues, then I think it means the default queue cannot be a plain queue either.

I also do not see why the default queue was made a separate from the destination queue (previously they were the same). Maybe that improves the test if events are dequeued only from the expected queue. But that seems like an unrelated change that should be in a different patch.

Reply:
> If plain queues are not supported as destination queues, then I think it means the default queue cannot be a plain queue either.

Updated the validation test to create default queue based on the new capability. On a related note, updated the API spec to say that the new queue type capability applies to both IPsec SA destination queue as well as the global IPsec default queue.

> I also do not see why the default queue was made a separate from the destination queue (previously they were the same). Maybe that improves the test if events are dequeued only from the expected queue. But that seems like an unrelated change that should be in a different patch.

Ack. Created a separate commit to introduce separate default queue into the test.

Reply:
I think the patches should be ordered differently. The set of patches that are directly related to the new queue type capability API change should come first. The other three patches (mode arg removal, separate default queue, time out) should come last. This way separate topics do not get mixed, making both the changes and the commit history more readable.

The other patches could have even been in a different PR (and we do not interleave commits of different PRs either, when merging), but it is ok to have them in this PR as well, after the API change related patches.

Reply:
> The other patches could have even been in a different PR (and we do not interleave commits of different PRs either, when merging), but it is ok to have them in this PR as well, after the API change related patches.

@JannePeltonen Bringing the default queue change after the API spec related changes looks challenging.

Once the queue type capability change is introduced in the validation test, I am forced to create and destroy the ipsec destination queue (suite_context.queue) per suite. But the "default_queue" passed to odp_ipsec_config() is global and cannot change per suite. So I am not able to pass the destination queue as default queue leading to segfault for lookup failure cases.

So in my opinion, the default queue related changes on validation test should come before the API spec related changes.

If you are OK with

1. moving the default queue related commits before API spec change commits in the same PR, I can do that way.
2. Or if you still prefer to create a separate PR for the default queue related changes, I can do that way.

Reply:
Both are fine. No need to create a new PR if you prefer that as I think this whole PR probably gets into a mergeable state soon.

The point was just to keep the API change unrelated improvements not interleaved with the dest_queue capability related stuff as I think that makes the individual patches easier to read. Whether they come before or after does not matter that much, so pick the order that you think makes sense.

Reply:
Ack. With v4, I have changed the order of the commits so that the API change related changes are not interleaved.